pkg/gadget-service/service-oci.go

```diff
@@ -35,6 +37,15 @@ func (s *Service) GetGadgetInfo(ctx context.Context, req *api.GetGadgetInfoReque
 		return nil, fmt.Errorf("expected version to be %d, got %d", api.VersionGadgetInfo, req.Version)
 	}
 
+	p, ok := peer.FromContext(ctx)
+	if ok {
+		tlsInfo, ok := p.AuthInfo.(credentials.TLSInfo)
+		if ok {
+			subject := tlsInfo.State.VerifiedChains[0][0].Subject
+			s.logger.Infof("[%s] GetGadgetInfo(%q)", subject, req.ImageName)
+		}
```

Comment:
No - it'll always be ok for TLS connections, as they're always verified. This will only be !ok if the connection is insecure (which is a valid and right now the default use case.)

Reply:
Hmm... It's pretty much a rabbit hole with the question "To what degree do we trust the imported fundamental packages like gRPC to work"? But if there's ever a good time to be paranoid, this might be it - hehe. I'll add that check.